### ISSUE TITLE
SCM Bitbucket - Permission Verification - Fails when more than 10 repo

### ISSUE THREAD

**Issue by xangxiong** [screwdriver-cd/screwdriver#1628]
**What happened**:
When using the Bitbucket SCM, if a user/team has more than 10 repository, the repository permission lookup will fail, resulting in pipeline creation failur due to user not having admin permission on repo.

**What you expected to happen**:
The permission verification should work even if the user has more than 10 repo.  The pipeline creation should finish successfully after being able to verify the permission.

**How to reproduce it**:
Configure the SCM Bitbucket and attempt to create a Pipeline with a Git repo that is part of a team/user with more than 10 other repo.

**Work Around**
I was able to implement a fix into my local screwdriver-scm-bitbucket/index.js file by adding the repo uuid to the getPerm url like:

```
 `${API_URL_V2}/repositories/${owner}/?q=uuid%3D%22${uuid}%22`
```

Also updating the role lookup to use '&' instead of '?'

```
if (desiredAccess === 'admin') {
    options.url = `${options.url}&role=admin`;
} else if (desiredAccess === 'push') {
    options.url = `${options.url}&role=contributor`;
} else {
    options.url = `${options.url}`;
}
```

**Comment by jithin1987** [comment]
@xangxiong  Thank you for the bug report. Glad that you were able to patch it locally. Would you be able to submit a patch for this ? https://github.com/screwdriver-cd/scm-bitbucket/

**Comment by xangxiong** [comment]
Thanks for the reply @jithin1987.  I can submit a patch.  Will post back once ready.

**Comment by xangxiong** [comment]
I have created a pull request here https://github.com/screwdriver-cd/scm-bitbucket/pull/58.  @jithin1987 , please review and see if we can merge it in.

**Comment by xangxiong** [comment]
Closing bug report.  This issue is fixed and is now part of tag release 3.4.3.
